src/scripts/macos/prepare-env.sh

```diff
@@ -8,6 +8,21 @@ readonly unity_editor_path="/Applications/Unity/Hub/Editor/$UNITY_EDITOR_VERSION
 printf '%s\n' "export UNITY_HUB_PATH=\"$unity_hub_path\"" >> "$BASH_ENV"
 printf '%s\n' "export UNITY_EDITOR_PATH=$unity_editor_path" >> "$BASH_ENV"
 
+check_and_install_rosetta() {
+  if ! /usr/bin/pgrep oahd &> /dev/null; then
+    echo "Rosetta 2 is not installed. Installing it now..."
+    softwareupdate --install-rosetta --agree-to-license
+    if [ $? -eq 0 ]; then
+      echo "Rosetta 2 installed successfully."
+    else
+      echo "Failed to install Rosetta 2."
+      exit 1
+    fi
+  else
+    echo "Rosetta 2 is already installed."
+  fi
+}
+
```

Comment:
We should keep an eye on updates from Unity for native ARM64 support, but I think this is the correct fix for this :+1: nice work 🙌